doc/tutorials/ann/ann.txt

```diff
@@ -0,0 +1,673 @@
+model.Add<Linear<> >(8, 3);
+model.Add<LogSoftMax<> >();
+
+// Train the model.
+model.Train(trainData, trainLabels);
+
+// Use the Predict method to get the assignments.
+arma::mat assignments;
+model.Predict(trainData, assignments);
+@endcode
+
+Now, the matrix assignments holds the classification of each point in the
+dataset.
+
+In the next example, we create simple noisy sine sequences, which are trained
+later on, using the RNN class.
+
+@code
+void GenerateNoisySines(arma::mat& data,
+                        arma::mat& labels,
+                        const size_t points,
+                        const size_t sequences,
+                        const double noise = 0.3)
+{
+  arma::colvec x =  arma::linspace<arma::Col<double>>(0,
+      points - 1, points) / points * 20.0;
+  arma::colvec y1 = arma::sin(x + arma::as_scalar(arma::randu(1)) * 3.0);
+  arma::colvec y2 = arma::sin(x / 2.0 + arma::as_scalar(arma::randu(1)) * 3.0);
+
+  data = arma::zeros(points, sequences * 2);
+  labels = arma::zeros(2, sequences * 2);
+
+  for (size_t seq = 0; seq < sequences; seq++)
+  {
+    data.col(seq) = arma::randu(points) * noise + y1 +
+        arma::as_scalar(arma::randu(1) - 0.5) * noise;
+    labels(0, seq) = 1;
+
+    data.col(sequences + seq) = arma::randu(points) * noise + y2 +
+        arma::as_scalar(arma::randu(1) - 0.5) * noise;
+    labels(1, sequences + seq) = 1;
+  }
+
+  const size_t rho = 10;
+
+  // Generate 12 (2 * 6) noisy sines. A single sine contains rho
+  // points/features.
+  arma::mat input, labelsTemp;
+  GenerateNoisySines(input, labelsTemp, rho, 6);
+
+  arma::mat labels = arma::zeros<arma::mat>(rho, labelsTemp.n_cols);
+  for (size_t i = 0; i < labelsTemp.n_cols; ++i)
+  {
+    const int value = arma::as_scalar(arma::find(
+        arma::max(labelsTemp.col(i)) == labelsTemp.col(i), 1)) + 1;
+    labels.col(i).fill(value);
+  }
+
+  /**
+   * Construct a network with 1 input unit, 4 hidden units and 10 output
+   * units. The hidden layer is connected to itself. The network structure
+   * looks like:
+   *
+   *  Input         Hidden        Output
+   * Layer(1)      Layer(4)      Layer(10)
+   * +-----+       +-----+       +-----+
+   * |     |       |     |       |     |
+   * |     +------>|     +------>|     |
+   * |     |    ..>|     |       |     |
+   * +-----+    .  +--+--+       +-----+
+   *            .     .
+   *            .     .
+   *            .......
+   */
+  Add<> add(4);
+  Linear<> lookup(1, 4);
+  SigmoidLayer<> sigmoidLayer;
+  Linear<> linear(4, 4);
+  Recurrent<> recurrent(add, lookup, linear, sigmoidLayer, rho);
+
+  RNN<> model(rho);
+  model.Add<IdentityLayer<> >();
+  model.Add(recurrent);
+  model.Add<Linear<> >(4, 10);
+  model.Add<LogSoftMax<> >();
+
+  StandardSGD opt(0.1, 1, input.n_cols /* 1 epoch */, -100);
+  model.Train(input, labels, opt);
+}
+@endcode
+
+For further examples on the ann usage of the ann classes, see [mlpack
+models](https://github.com/mlpack/models)
```

Comment:
This is a tiny issue but probably we should put a period at the end of this sentence.  I think "ann usage" should be just "usage" also. :)